### ISSUE TITLE
Fix linting issues caused due to  Loop With Too Many Jump Statements as detected by detekt

### ISSUE THREAD

**Issue by angrezichatterbox** [scribe-org/Scribe-Android#91]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Android/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Android's [Code of Conduct](https://github.com/scribe-org/Scribe-Android/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Loops with too many break, continue, or return statements can make the code harder to follow.

To fix this issue the loops have to be simplified to reduce the number of breaking statements like if conditions. 

Example:
Before Fixing:
```
fun processItems(items: List<Int>) {
    for (item in items) {
        if (item == 0) break // Too many jump statements in this loop
        if (item % 2 == 0) continue
        println(item)
    }
}
```

After fixing:

```
fun processItems(items: List<Int>) {
    for (item in items.filter { it != 0 && it % 2 != 0 }) {
        println(item)  // Simplified logic with fewer jump statements
    }
}

```
More of this could be found from the [docs](https://detekt.dev/docs/rules/style/#loopwithtoomanyjumpstatements)
These can be identified across the code by the following steps:

- Enable the LoopWithTooManyJumpStatements rule in detekt.yml.
- Run ./gradlew detekt to detect such loops.
- Refactor the identified loops to reduce complexity.

### Contribution

I would love to work on this and am more than willing to help anyone solve this issue. 😄

**Comment by angrezichatterbox** [comment]
I will be working on this issue :)

**Comment by andrewtavis** [comment]
Thanks for picking this up, @angrezichatterbox!

**Comment by andrewtavis** [comment]
Closed by #238 🚀 Thanks @angrezichatterbox!
